### ISSUE TITLE
Autocomplete should allow arbitrary suggestions (even not matching user input)

### ISSUE THREAD

**Issue by valerio-bozzolan** [Dogfalo/materialize#6420]
## Expected Behavior
Hello! When using the autocomplete with the real world we should allow to add at runtime also arbitrary autocomplete suggestions (that does not match literally the user input). For exampe, when I search "intelligent", a valid suggestion may be "stupid" (what if I'm a dictionary of Contraries and Synonyms? who knows!).

So this library should not force a literal search engine.

Actually I think it's not possible, looking at my source code of Materialize v1.0.0, because the search logic is inside the `_renderDropdown()` part.

## Current Behavior
When the user type "soccer" and you want to show an arbitrary suggestion as, for example, "football", actually I think you can't without workarounds like suggestions like "soccer (football)".

## Possible Solution
Add an option (like `disableInternalSearch` or whatever) to skip the internal checks and show whatever data. In this way, as default the users can have an empty set and just run `updateData()` at runtime to show these data as-is, without other unwanted literal filters.

Another approach could be a custom callback for the filter logic, but for my cause it would be just a stupid:

```
whateverOptionNameFilter: function(query, tip) {
   return true;
}
```

Because the business logic is not in JavaScript but it's server side.

Anyway, both solutions may be good IMVHO.

## Your Environment
Actually to retrieve the suggestions and display them I'm listening to the `input` event on the `autocomplete.el` to run my Ajax requests (when the user stop typing). Then, with the Ajax response I populate the autocomplete with `autocomplete.updateData()` +`autocomplete.show()`, but I have to assure that the user input is __inside__ my autocomplete suggestion entries, or actually I can't figure out how let them be displayed.

**Comment by ccssmnn** [comment]
I was looking for a similar feature. In particular, I wanted to use string similarity algorithms to find and sort matches. Since you can already provide a custom **sortFunction** how about providing a custom **matchFunction**?

You can provide that feature in your autocomplete with inheriting from M.Autocomplete and overriding _renderDropdown. [Here](https://gist.github.com/ccssmnn/9324542206df72731bdadbf477ca5cdd) is a gist, but this is rather ugly.

**Comment by valerio-bozzolan** [comment]
Probably a duplicate of #6245

**Comment by acburst** [comment]
Merged #6245
